cli/src/commands/version.rs

```diff
@@ -0,0 +1,40 @@
+use std::process::Command;
+
+use clap::Parser as ClapParser;
+use log::warn;
+
+/// Prints version information
+///
+/// This subcommand is implemented for backward compatibility reasons.
+/// Its behavior should be the same as that of the `--version` flag which
+/// is handled internally by clap.
+#[derive(Debug, ClapParser)]
+pub struct Opts {}
+
+pub fn run(_opts: Opts) {
+    let commit = match Command::new("git")
+        .args(["rev-parse", "--short=11", "HEAD"])
+        .output()
```

Comment:
The issue I have here is that this will only work if you're within the right directory.

Having the igt hash is v useful though; Perhaps we should have a `build.rs` thing to get it compiled into the output binary as a static string that this can reference?

Reply:
Actually https://stackoverflow.com/questions/43753491/include-git-commit-hash-as-string-into-rust-program looks interesting.

Inparticular the https://github.com/rustyhorde/vergen crate, which ensures we'll re-run in the right cases. Though I also wouldn't be opposed to a `build.rs` that manually emits a rerun-if-changed for `.git/HEAD` and uses `.git/HEAD` and then whatever file that points to to find the current git commit hash and put it into an env variable for the program to pull out.